flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala

```diff
@@ -498,6 +505,10 @@ abstract class TableEnvironment(val config: TableConfig) {
     TableEnvironment.validateType(inputType)
```

Comment:
Can we do the `GenericTypeInfo` type check in `TableEnvironment.validateType(inputType)` method? If we do so, we can remove the duplicate check. What do you think?
Best,
SunJincheng